File: LeetCode/1572. Matrix Diagonal Sum/solution.cpp
/**
 * Runtime: 8 ms
 * Memory Usage: 10.9 MB 
 */
class Solution {
public:
    int diagonalSum(vector<vector<int>>& mat) {
        int n = mat.size(), mid = n >> 1;
        int ans = -mat[mid][mid] * (n & 1);
        for (int i = 0; i < n; ++ i) {
            ans += mat[i][i] + mat[i][n - 1 - i];
        }
        return ans; 
    }
};

File: LeetCode/1572. Matrix Diagonal Sum/solution.py
"""
Runtime: 32 ms
Memory Usage: 15.2 MB
"""
class Solution:
    def diagonalSum(self, mat: List[List[int]]) -> int:
        n = len(mat)
        mid= n >> 1
        ans = -mat[mid][mid] * (n & 1)
        for i in range(n):
            ans += mat[i][i] + mat[i][n - 1 - i]
        return ans
